frontend/src/app/components/admin/user-index/admin-user-index/admin-user-show.component.ts

```diff
@@ -1,37 +1,66 @@
 import { Component, OnInit } from '@angular/core';
 import { MatTableDataSource, MatFormFieldControl } from '@angular/material';
-import { UserService } from 'src/app/services/user.service';
-
-export interface UserElement {
-  name: string;
-  dob: string;
-  position: string;
-  supervisor: string;
-  doe: string;
-  email: string;
-}
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA} from '@angular/material/dialog';
+import { CreateUserComponent } from 'src/app/modals/create-user/create-user.component';
+import { User } from 'src/app/models/User.model';
+import { ApiCommunicationService } from 'src/app/services/ApiCommunication.service';
+import { Group } from 'src/app/models/Group.model';
 
-const Data: UserElement[] = [
-  {name: 'employee', dob: '1990-10-11', position: 'developer', supervisor: 'ceo', doe: '2019-10-10', email: 'emp@samplemail.com'},
-  {name: 'employeenr2', dob: '1995-10-11', position: 'staff', supervisor: 'ceo', doe: '2018-10-10', email: 'employee@samplemail.com'}
-]
 
 @Component({
   selector: 'app-admin-user-show',
   templateUrl: './admin-user-show.component.html',
   styleUrls: ['./admin-user-show.component.css']
 })
 export class AdminUserShowComponent implements OnInit {
-
   displayedColumns: string[] = ['name', 'dob', 'position', 'supervisor', 'doe', 'email'];
-  dataSource = new MatTableDataSource(Data); // --> filter
-  
+  dataSource; // --> filter
+
+  userData: User;
+  usersList: User[];
+
+  constructor(private api: ApiCommunicationService, public dialog: MatDialog) {}
 
+  createUser(): void {
+    const dialogRef = this.dialog.open(CreateUserComponent, {
+      data: {user: this.userData}
+    });
 
-  constructor(private userService: UserService) { }
+    dialogRef.afterClosed().subscribe(result => {
```

Comment:
it might needs an unsubscribe

Reply:
it doesn't work with unsubscribe.

Reply:
declare a variable just before the constructor
`private unsubscribe$ = new Subject<void>();`

Implement ng on destory
`	ngOnDestroy(): void {
		this.unsubscribe$.next();
		this.unsubscribe$.complete();
	}`

Insert a pipe before subscribe with the unsubscribe logic

```suggestion
    dialogRef.afterClosed().pipe(takeUntil(this.unsubscribe$)).subscribe(result => {
```

https://alligator.io/angular/takeuntil-rxjs-unsubscribe/